apps/common/models/i-casework-getcase.js

```diff
@@ -29,9 +28,16 @@ module.exports = class DocumentModel extends Model {
   }
 
   fetch() {
-    const options = this.requestConfig({});
-    options.qs = this.prepare();
-    options.method = 'GET';
-    return this.request(options);
+    const params = {
+      url: this.url(),
+      method: 'GET',
+      params: this.prepare()
+    };
+    return this._request(params).then(response => {
+      return this.parse(response.data);
+    })
+      .catch(err => {
+        return err;
```

Comment:
```suggestion
        console.error(`Error fetching data from ${params.url}: ${err.message}`);
        throw new Error(`Failed to fetch data: ${err.message || 'Unknown error'}`);
```

Reply:
obviously do it with the logger if you can

Reply:
I hsve added in the logger now